Submitted By: Jim Gifford (patches at jg555 dot com)
Date: 2006-01-09
Initial Package Version: Snapshot
Origin: Glibc CVS and Jim Gifford
Upstream Status: Already Applied
Description: Fixed pthreadtypes for 20060109 snapshot of glibc
 
diff -Naur glibc-20060109.orig/nptl/sysdeps/unix/sysv/linux/mips/bits/pthreadtypes.h glibc-20060109/nptl/sysdeps/unix/sysv/linux/mips/bits/pthreadtypes.h
--- glibc-20060109.orig/nptl/sysdeps/unix/sysv/linux/mips/bits/pthreadtypes.h	2005-03-29 19:56:26.000000000 +0000
+++ glibc-20060109/nptl/sysdeps/unix/sysv/linux/mips/bits/pthreadtypes.h	2006-01-10 05:15:57.000000000 +0000
@@ -1,5 +1,5 @@
 /* Machine-specific pthread type layouts.  MIPS version.
-   Copyright (C) 2005 Free Software Foundation, Inc.
+   Copyright (C) 2005 2006 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -59,7 +59,7 @@
    type is deliberately not exposed.  */
 typedef union
 {
-  struct
+  struct  __pthread_mutex_s
   {
     int __lock;
     unsigned int __count;
@@ -70,10 +70,19 @@
     /* KIND must stay at this position in the structure to maintain
        binary compatibility.  */
     int __kind;
-#if _MIPS_SIM != _ABI64
+#if _MIPS_SIM == _ABI64
+    int __spins;
+    struct __pthread_mutex_s *__next;
+    struct __pthread_mutex_s *__prev;
+# define __PTHREAD_MUTEX_HAVE_PREV      1
+#else
     unsigned int __nusers;
+    union
+    {
+      int __spins;
+      struct __pthread_mutex_s *__next;
+    };
 #endif
-    int __spins;
   } __data;
   char __size[__SIZEOF_PTHREAD_MUTEX_T];
   long int __align;
